test/util/util.go

```diff
@@ -208,21 +214,20 @@ func NewClient(serverUrl string, caCert *crypto.TLSCertificateConfig) (*client.C
 }
 
 // NewClient creates a new client with the given server URL and certificates. If the certs are nil a http client will be created.
-func NewAgentClient(serverUrl string, caCert, clientCert *crypto.TLSCertificateConfig) (*agentclient.ClientWithResponses, error) {
-	httpClient, err := NewBareHTTPsClient(caCert, clientCert)
+func NewAgentClient(serverUrl string, caBundle []*x509.Certificate, clientCert *crypto.TLSCertificateConfig) (*agentclient.ClientWithResponses, error) {
+	httpClient, err := NewBareHTTPsClient(caBundle, clientCert)
 	if err != nil {
 		return nil, fmt.Errorf("creating TLS config: %v", err)
 	}
 
 	return agentclient.NewClientWithResponses(serverUrl, agentclient.WithHTTPClient(httpClient))
 }
 
-func NewBareHTTPsClient(caCert, clientCert *crypto.TLSCertificateConfig) (*http.Client, error) {
+func NewBareHTTPsClient(caBundle []*x509.Certificate, clientCert *crypto.TLSCertificateConfig) (*http.Client, error) {
 
 	httpClient := &http.Client{}
-	if caCert != nil || clientCert != nil {
-		var err error
-		tlsConfig, err := crypto.TLSConfigForClient(caCert, clientCert)
+	if len(caBundle) > 0 || clientCert != nil {
```

Comment:
should this be and vs or?

Reply:
IMHO it is correct. 
The way test are set up the connection is moved to TLS mode if you have either a CA bundle or a client cert mTLS.
It is an equivalent of the old condition (see the diff). caCert != nil is one or more certs in the bundle concatenated as PEM.